### DISCUSSION TITLE
Pass arguments to VSCodeNotify

### DISCUSSION THREAD

**Discussion by ryands17** [vscode-neovim/vscode-neovim#1934] [Q&A] [answered]
Is it possible to pass arguments to `VSCodeNotify` when calling an operation? I am using it to call a workbench command and I need it to only be fired on specific conditions i.e. adding a `when` clause.

**Comment by justinmk** [comment, marked as answer]
see https://github.com/vscode-neovim/vscode-neovim?tab=readme-ov-file#vimscript which links to https://github.com/vscode-neovim/vscode-neovim?tab=readme-ov-file#%EF%B8%8F-api which shows examples passing args.

**Reply by ryands17** [reply]
I tried passing it in like this but it doesn't seem to work:
```lua
args = {
  when = "!notebookEditorFocused && editorLangId == 'markdown'"
}
```

I'm actually trying to emulate the VSCode `when` parameter in the keybindings

**Reply by justinmk** [reply]
what vscode command accepts a "when" parameter? typically `when` is specified directly in settings/keybindings json. if there is a vscode command that accepts a `when` param, then a reference would help. also showing a full code sample.

**Reply by ryands17** [reply]
I usually set a `when` condition in VSCode in the `keybindings.json` file like this:
```json
{
    "key": "r",
    "command": "renameFile",
    "when": "filesExplorerFocus && foldersViewVisible && !explorerResourceIsRoot && !explorerResourceReadonly && !inputFocus"
}
```

Is this possible to be sent down in the args keyword? I am actually setting the keybinding in my neovim config so I don't know how to connect it to the `when` option in VSCode.

**Reply by justinmk** [reply]
> Is this possible to be sent down in the args keyword?

I have no idea. That's a question about the vscode API, which is unrelated to the general topic of "how to send args using VSCodeNotify"

**Comment by theol0403** [comment]
When is not an argument for a command, it's a directive form the keybindings parser.
